services/exportService.ts

```diff
@@ -0,0 +1,75 @@
+import { getSiteFeeds, insertSiteFeed } from "../services/indexeddbService";
+import { SiteFeed, TransferData, TransferFeed } from "../models";
+import { databaseVersion } from "../constants";
+import { compress, decompress}  from "./compressService";
+import { siteDomain } from "../constants";
+
+export async function exportFeed(): Promise<string | undefined>{
+
+    let transferFeed: TransferFeed[] = [];
+    const feedStore = await getSiteFeeds() as SiteFeed[];
+    feedStore.forEach(feed => transferFeed.push(refactorFeedToExport(feed)))
+    
+    const transferData: TransferData = {db: databaseVersion, feed: transferFeed}; 
+
+    const compressedData: string =  compress(JSON.stringify(transferData));
+    const linkToExport = siteDomain + compressedData;
+
+    // method to download data as .* file
+    // if(linkToExport.length > 2048){
+    //     return;
+    // }
+
+    return linkToExport;
+}
+
+export async function importFeed(link: string){
+
+    const transferString = decompress(link.replace(siteDomain, ""));
+
+    const transferData = JSON.parse(transferString) as TransferData;
+
+    if(transferData.feed){
+        transferData.feed.forEach(feed => insertSiteFeed(refactorExportToFeed(feed)));
+        console.log("Database updated successfully");
+    }
+    else{
+        alert("Feed is empty")
+    }
+}
+
+function refactorExportToFeed(feed: TransferFeed): SiteFeed{
+
+    const url: string = feed.domain + feed.url;
+
+    let visited: Record<string, boolean> = {};
+    feed.visited?.forEach(path => visited[path] = true)
```

Comment:
We can use here reduce:
```suggestion
    const visited: Record<string, boolean> = feed.visited?.reduce((acc,path) => {
      acc[feed.domain + path] = true;
      return acc;
     }, {}) || {};
```